### ISSUE TITLE
Can see any tokens with just a few pixels of visibility

### ISSUE THREAD

**Issue by Cubius** [dev7355608/perfect-vision#222]
I can see any tokens so long as there is just the tiniest sliver of their tile exposed.

![image](https://user-images.githubusercontent.com/15717384/156246431-1e6a90e6-106d-44e2-ad4a-a3eb1e80aade.png)

**Comment by dev7355608** [comment]
This is an intentional improvement over core Foundry. Core Foundry just test a couple of points near the center of the token, which leads to token being invisible even if ~20% of them should be visible (this is especially noticeable with tokens larger than one square). Perfect Vision tests the exact circle of the token. There's also a sight limit related reason why PV introduced this improved test (see #177).

**Comment by Cubius** [comment]
#177 mentions tokens not being seen with just a sliver of them technically exposed as proper behaviour, they seemed mainly concerned with the token being able to see out.
Might I suggest two new settings to enable/disable each - seeing tokens and seeing from points other than the center of the token.
Having tokens not show through some not perfectly grid aligned walls or doors is a very nice quality of life feature.
Perhaps move the circle of points to be checked half a tile inward from a token's rim if that is a possibility? That would reduce small tokens back to a point but make larger tokens still more perceptible but not risk their rims poking through walls either.

**Comment by Cubius** [comment]
>>Perhaps move the circle of points to be checked half a tile inward from a token's rim if that is a possibility? That would reduce small tokens back to a point but make larger tokens still more perceptible but not risk their rims poking through walls either.

> That would be pretty much like core.

Not quite, on 1 tile tokens it would be core behaviour, but larger ones (afaik) still only check for the token's center point.
Setting a radius that grows relative to token size could mitigate the issue of not being able to see a token while the center is obscured on larger tokens where this is an issue.
![image](https://user-images.githubusercontent.com/15717384/156369072-c3a0b2c2-e002-4ce1-9b3d-9885b7eb8c90.png)

> I think there's a module out there that allows to change to the sight origin, I think it's called Point of View, but it might be no longer maintained.

I found [this module](https://github.com/trioderegion/eagle-eye), which uses 4 sight origins on the corners of a token's tile, though whether that is desirable behaviour is an entirely new topic of discussion.

> What do you mean by "seeing tokens" precisely?

I meant the method by which tokens are being revealed to an observing token - one point in the center or a larger radius.

**Comment by Shuggaloaf** [comment]
> ...tokens not show [im]perfectly aligned walls

> you're the first one to complain about it.

I came here to make a new suggestion and just happened to see this issue. I mention this because I have noticed this issue as well and I honestly thought it was core behavior (have had the mod for several Foundry versions now, so been a while since I saw non-PV/core behavior)

So this could be the same with others. Maybe like me they noticed it, thought it was core or unsure of cause, decided not a big enough deal to really dig into it. I never would have known this was PV behavior without just randomly seeing this issue here. 

I get that most of that is a "me issue", but just trying to add a little perspective.
